src/redux/reducers/index.js

```diff
@@ -88,10 +83,17 @@ function notes(
       return Object.assign({}, state, {
         allNotes: updateNoteInAllNotes(action.note, state.allNotes)
       });
-    case DELETE_NOTE:
+    case DELETE_NOTE: {
```

Comment:
The only concern here is that deleting a note and selecting some other one are now coupled. The action is simply called `DELETE_NOTE`, but it does more than that, so that is kinda deceptive. What if you wanted to create a new action that deleted a note but then selected a note based on a different method? I would break down this action like it was and simply create an action that dispatches both a delete and select some note action.

Reply:
I think creating an action that dispatches both delete and select some note is the best choice here. I combined these specifically so that they would be atomic. Or so that you literally could not delete a note without selecting something. Deleting without selecting can cause weird issues and someone unfamiliar with the code might not realize they need to both delete AND do some other action.

Reply:
Yeah, so break down this redux reducer code, but create an action that does both.